Gems/PhysX/Code/Editor/DebugDraw.cpp

```diff
@@ -687,6 +687,53 @@ namespace PhysX
             [[maybe_unused]] const AZ::Vector3& colliderScale,
             [[maybe_unused]] const bool forceUniformScaling) const
         {
+            const int numColumns = heightfieldShapeConfig.GetNumColumns();
+            const int numRows = heightfieldShapeConfig.GetNumRows();
+
+            const float minXBounds = -(numColumns * heightfieldShapeConfig.GetGridResolution().GetX()) / 2.0f;
+            const float minYBounds = -(numRows * heightfieldShapeConfig.GetGridResolution().GetY()) / 2.0f;
+
+            auto heights = heightfieldShapeConfig.GetSamples();
+            
+            for (int xIndex = 0; xIndex < numColumns - 1; xIndex++)
+            {
+                for (int yIndex = 0; yIndex < numRows - 1; yIndex++)
+                {
+                    const int index0 = yIndex * numColumns + xIndex;
+                    const int index1 = yIndex * numColumns + xIndex + 1;
+                    const int index2 = (yIndex + 1) * numColumns + xIndex;
+                    const int index3 = (yIndex + 1) * numColumns + xIndex + 1;
+
+                    const float x0 = minXBounds + heightfieldShapeConfig.GetGridResolution().GetX() * xIndex;
+                    const float x1 = minXBounds + heightfieldShapeConfig.GetGridResolution().GetX() * (xIndex + 1);
+                    const float y0 = minYBounds + heightfieldShapeConfig.GetGridResolution().GetY() * yIndex;
+                    const float y1 = minYBounds + heightfieldShapeConfig.GetGridResolution().GetY() * (yIndex + 1);
```

Comment:
`heightfieldShapeConfig.GetGridResolution()` can be saved in a const variable outside the loop and used in the calculation of `minXBounds`, `minYBounds`, `x0`, `x1``,  `y0`,  `y1`.

Reply:
Done